src/main/java/de/hysky/skyblocker/skyblock/speedPreset/SpeedPresetListWidget.java

```diff
@@ -0,0 +1,208 @@
+package de.hysky.skyblocker.skyblock.speedPreset;
+
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.DrawContext;
+import net.minecraft.client.gui.Drawable;
+import net.minecraft.client.gui.Element;
+import net.minecraft.client.gui.Selectable;
+import net.minecraft.client.gui.widget.*;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class SpeedPresetListWidget extends ElementListWidget<SpeedPresetListWidget.AbstractEntry> {
+
+	private static final Pattern NUMBER = Pattern.compile("^-?\\d+(\\.\\d+)?$");
+	// Alphanumeric sequence that doesn't start with a number.
+	private static final Pattern TITLE = Pattern.compile("^[a-zA-Z][a-zA-Z0-9_]*$");
+	private static final Logger LOGGER = LoggerFactory.getLogger(SpeedPresetListWidget.class);
+
+	public SpeedPresetListWidget(int width, int height, int y) {
+		super(MinecraftClient.getInstance(), width, height, y, 25);
+		var presets = SpeedPresets.getInstance();
+		addEntry(new TitleEntry());
+		if (presets.getPresetCount() > 0)
+			presets.forEach((title, speed) ->
+					this.addEntry(new SpeedPresetEntry(title, String.valueOf(speed))));
+		else
+			this.addEntry(new SpeedPresetEntry("", ""));
+	}
+
+	@Override
+	public int getRowWidth() {
+		return super.getRowWidth() + 104;
+	}
+
+	/**
+	 * Checks whether the state of the widget has been modified compared to the preset data.
+	 * <p>
+	 * This method determines if any of the following conditions are met:
+	 * <ul>
+	 *     <li>The number of children entries in the widget is less than the current number of presets.</li>
+	 *     <li>An entry has been modified.</li>
+	 *     <li>The number of entries worth saving does not match the number of presets.</li>
+	 * </ul>
+	 *
+	 * @return true if the state has been modified; false otherwise.
+	 */
+	public boolean hasBeenChanged() {
+		var presets = SpeedPresets.getInstance().getPresetCount();
+		// Accounting 1 for the title entry.
+		var children = children().size() - 1;
+		if (children < presets) return true;
+
+		children = 0;
+		for (var child : children()) {
+			if (child.hasBeenModified()) return true;
+			if (child instanceof SpeedPresetEntry entry) {
+				children = entry.isEmpty() ? children : children + 1;
+			}
+		}
+		return children != presets;
```

Comment:
```suggestion
		return children().stream().anyMatch(AbstractEntry::hasBeenModified) || children().stream().filter(SpeedPresetEntry.class::isInstance).map(SpeedPresetEntry.class::cast).filter(entry -> !entry.isEmpty()).count() != SpeedPresets.getInstance().getPresetCount();
```

Reply:
This results into two separate loops although it could be done it one